src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/RazorDocumentFormattingEndpoint.cs

```diff
@@ -63,7 +55,7 @@ public TextDocumentIdentifier GetTextDocumentIdentifier(DocumentFormattingParams
             return null;
         }
 
-        var documentContext = await _documentContextFactory.TryCreateAsync(request.TextDocument.Uri, cancellationToken).ConfigureAwait(false);
```

Comment:
I don't know how this got through my CLaSP scrub but (almost) no endpoint should have the DocumentContextFactory. If you want the DocumentContext get it off the `RazorRequestContext`. To do it this way is a source of possible race conditions.

AFAIK there's one exception for Rename (because it needs the context for multiple documents), if anyone thinks there's another endpoint which needs to be exceptional lets have a chat.

Reply:
Agreed: https://github.com/dotnet/razor/issues/8126.